refactor!: Use circuit builders

codygunton opened the pull request:
# Description

This PR is 'just' bulk cleanup to reflect changes in Barretenberg. These are the main changes.
 - `Composer` is renamed to `Builder` or `CircuitBuilder`
 - `composer` is renamed to `builder`. In some contexts `circuit` also makes sense and it might be preferable.
 - Circuit builders are not instantiated with a string specifying a crs, so these constructors (which were available because of a hack) are removed.
 - `ComposerType` was renamed to `CircuitType` and this is changed accordingly, along with renaming `composer_type` to `circuit_type`.
 - Rename `DummyComposer` and dummy_composer.hpp.
 - In order to restrict the barretenberg API, the `Aggregator::aggregate` function now calls a more encapsulated recursive verifier function.

Some unexpected compiler errors around unused variables came up. I'm not sure how to proceed. I made an issue (#914) recording this in the hopes that we can merge with todos referencing that issue. 

I left some additional notes as comments on lines of code.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] The branch has been merged or rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.

codygunton committed:
Use circuit constructor.

codygunton committed:
Undo hack to stop exposing manifest.

codygunton committed:
Update Bb

codygunton committed:
Update Bb

codygunton committed:
Bump Bb (need Mont conversion)

ludamad commented on circuits/cpp/src/aztec3/circuits/kernel/public/.test.cpp:
```diff
@@ -361,12 +361,12 @@ PublicKernelInputs<NT> get_kernel_inputs_with_previous_kernel(NT::boolean privat
-                                                                    .l1_to_l2_messages_tree_root = 3000,
-                                                                    .private_kernel_vk_tree_root = 4000,
-                                                                } };
+    // CombinedHistoricTreeRoots<NT> const historic_tree_roots = { .private_historic_tree_roots = {
```
Should we remove instead?

codygunton replied:
Probably? I haven't looked into the history or discussed with core circuits team yet.

ludamad replied:
Yeah I didn't see it was a draft, my bad

codygunton replied:
I ended up making an issue to look at this and two other unused variables in the same test file, and I reference that issue in TODOs in the code.

ludamad reviewed (approved):
LGTM overall (although now I realize its a draft)

codygunton committed:
Using builder and CircuitType

codygunton committed:
Fix some things break others.

codygunton committed:
Satisfy prettier?

codygunton committed:
Composer ~> Builder

codygunton committed:
composer ~> builder

codygunton committed:
mv dummy_composer.hpp and rename class.

codygunton committed:
Clean change to settingss.

codygunton committed:
Bump Bb.

codygunton committed:
Merge branch 'master' into cg/circuit-builders
Weird stuff is happening

codygunton committed:
Note on compilation errors in tests.

codygunton committed:
Try to satisfy clang-tidy.

codygunton committed:
Fix test that didn't compile before.

suyash67 commented on circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit_ordering.hpp:
```diff
@@ -7,26 +7,26 @@
         read_request_membership_witnesses);
 
-CircuitResult<KernelCircuitPublicInputs<NT>> native_private_kernel_circuit_ordering_rr_dummy(
+CircuitResult<KernelCircuitPublicInputs<NT> > native_private_kernel_circuit_ordering_rr_dummy(
```
nit but is this an unwanted space?

codygunton replied:
Fixed

suyash67 commented on circuits/cpp/src/aztec3/circuits/kernel/private/private_kernel_circuit.cpp:
```diff
@@ -27,22 +27,16 @@ using aztec3::circuits::silo_nullifier;
     // compute P0, P1 for private function proof
-    CT::AggregationObject aggregation_object = Aggregator::aggregate(
-        &composer, private_inputs.private_call.vk, private_inputs.private_call.proof, num_private_call_public_inputs);
+    CT::AggregationObject aggregation_object =
```
This is nice, `aggregate` shouldn't have to know what the number of public inputs of the private call is.

suyash67 commented on yarn-project/circuits.js/src/structs/verification_key.ts:
```diff
@@ -1,7 +1,7 @@
 import { serializeToBuffer } from '../utils/serialize.js';
 import { Fq } from './index.js';
-import { ComposerType } from './shared.js';
+import { CircuitType } from './shared.js';
```
Looks good.

suyash67 commented on circuits/cpp/src/aztec3/utils/types/circuit_types.hpp:
```diff
@@ -9,43 +9,40 @@ using namespace proof_system::plonk;
-        std::conditional_t<std::same_as<Composer, plonk::TurboPlonkComposer>,
-                           stdlib::recursion::recursive_turbo_verifier_settings<bn254>,
-                           stdlib::recursion::recursive_ultra_verifier_settings<bn254>>;
+    using recursive_inner_verifier_settings = stdlib::recursion::recursive_ultra_verifier_settings<bn254>;
```
Is his because we don't expect turbo to be used as inner circuit builder anymore? If so, makes sense.

codygunton replied:
Well, Luke's previous PR fixed the inner circuit type to Ultra (indeed, `Aggregate::aggregate` now uses a fixed flavor), so it wouldn't even work to toggle the inner type. In general this recursion stuff needs a rework, so I thought I'd pare it down to make it simpler for when we do that work. In fact, I see now this alias is not in use anywhere, so I think I'll remove it.

suyash67 reviewed (approved):
Looks good, just left a couple of minor comments. Thanks @codygunton.

codygunton committed:
Merge branch 'master' into cg/circuit-builders

codygunton committed:
Respond to comments and clean up

codygunton committed:
Fix typo and formatting.

codygunton committed:
Merge branch 'master' into cg/circuit-builders

codygunton committed:
Fix compilation post-merge

codygunton committed:
It should be a Schnorr signature?

codygunton committed:
Merge branch 'master' into cg/circuit-builders

codygunton committed:
Revert to Ecdsa

codygunton committed:
Proposed fix.

codygunton committed:
Revert "Proposed fix."

This reverts commit bef4f57a5ec69deeeca1b1b39bd95eb6118a7302.

codygunton committed:
Test Schnorr; update snapshot.

Changing test description to reference Schnorr not Ecdsa leads to weird build error.

codygunton committed:
Rebuild WASM before making snapshot.

codygunton commented on yarn-project/circuits.js/src/structs/kernel/__snapshots__/index.test.ts.snap:
```diff
@@ -113,7 +113,7 @@ value: 0xd05
 "
 `;
 
-exports[`structs/kernel serializes and prints EcdsaSignature 1`] = `"{ 0101010101010101010101010101010101010101010101010101010101010101, 0101010101010101010101010101010101010101010101010101010101010101, 0 }"`;
```
As discussed with @PhilWindle, this final value in the snapshot here should not exist because the signature tested is not an ECDSA signature as the naming suggests--it was switched in some places to a Schnorr signature. I complete this switch by changing the definition of `abis__test_roundtrip_serialize_signature` to use a Schnorr signature and updated the snapshot accordingly.

codygunton replied:
BTW, I tried to rename the test and ran into a weird error:
```
 FAIL  structs/kernel/index.test.ts
  ● Test suite failed to run

    ReferenceError: exports is not defined

      1 | import { expectSerializeToMatchSnapshot } from '../../tests/expectSerialize.js';
    > 2 | import { makeSchnorrSignature } from '../../tests/factories.js';
        |                       ^
      3 | import {
      4 |   makePreviousKernelData,
      5 |   makePrivateKernelInputsInner,

      at structs/kernel/index.test.ts:2:23
```
